InterGalacticCurrencyConvertor/src/main/java/numerals/RomanNumerals.java

```diff
@@ -0,0 +1,145 @@
+		int vMinimum = vBefore;
+		Integer vRiseOcc = -1;
+		String vCurRepRN = inRNums.get(0);
+		for (int i = 1; i < inRNums.size(); ++i) {
+			if (!isRomanNumeral(inRNums.get(i))) {
+				return vResult;
+			}
+			vAfter = getDecimalVal(inRNums.get(i));
+			vOccurrenceMap.put(inRNums.get(i), vOccurrenceMap.get(inRNums.get(i)) + 1);
+
+			if (vBefore < vAfter) {
+				vRiseOcc = i;
+				int j = i - 2, count = 1;
+				while (j >= 0) {
+					if (getDecimalVal(inRNums.get(j)) < vAfter){
+						++count;
+					}
+					if (count > 1) {
+						return vResult;
+					}
+					--j;
+				}
+				if (vCurRepRNCnt > 1 && !(inRNums.get(i-1).equals("i") ||
+				    inRNums.get(i-1).equals("x") || inRNums.get(i-1).equals("c") ||
+				    inRNums.get(i-1).equals("m"))) {
+					return vResult;
+				} else if (inRNums.get(i-1).equals("i") && !(inRNums.get(i).equals("v") ||
+				           inRNums.get(i).equals("x"))) {
+					return vResult;
+				} else if (inRNums.get(i-1).equals("x") && !(inRNums.get(i).equals("l") ||
+				           inRNums.get(i).equals("c"))) {
+					return vResult;
+				} else if (inRNums.get(i-1).equals("c") && !(inRNums.get(i).equals("d") ||
+				           inRNums.get(i).equals("m"))) {
+					return vResult;
+				} else if (inRNums.get(i-1).equals("v") || inRNums.get(i-1).equals("l") ||
+				           inRNums.get(i-1).equals("d")){
+					return vResult;
+				}
+				//System.out.println(inRNums.get(i-1) + " rise " + inRNums.get(i));
+			}
+			if (vRiseOcc > -1 && vRiseOcc < i && vMinimum < getDecimalVal(inRNums.get(i))) {
+				return vResult;
+			}
+			if (inRNums.get(i) != vCurRepRN) {
+				vCurRepRN = inRNums.get(i);
+				vCurRepRNCnt = 1;
+			} else {
+				++vCurRepRNCnt;
+			}
+			//System.out.println("middle");
+			if (vCurRepRNCnt > 1) {
+				if (!(inRNums.get(i).equals("i") || inRNums.get(i).equals("x") ||
+				    inRNums.get(i).equals("c") || inRNums.get(i).equals("m"))) {
+					return vResult;
+				} else if (vCurRepRNCnt == 4 && inRNums.get(i).equals(vCurRepRN)) {
+					return vResult;
+				}
+			} else if (vOccurrenceMap.get("i") > 4 || vOccurrenceMap.get("x") > 4 ||
+			           vOccurrenceMap.get("c") > 4 || vOccurrenceMap.get("m") > 4 ||
+			           vOccurrenceMap.get("v") > 1 || vOccurrenceMap.get("l") > 1 ||
+			           vOccurrenceMap.get("d") > 1) {
+				return vResult;
+			}
+
+			vBefore = vAfter;
+			if (vMinimum > getDecimalVal(inRNums.get(i))) {
+				vMinimum = getDecimalVal(inRNums.get(i));
+			}
+			//System.out.println("end " + vCurRepRNCnt);
+		}
+		vResult = true;
+		return vResult;
+	}
+	public boolean isRomanNumeral(String inRNum) {
+		if (inRNum == null) {
+			return false;
+		}
+		return mMap.containsKey(inRNum);
+	}
+	public Integer getDecimalVal(String inRNum) {
+		if (!isRomanNumeral(inRNum)) {
+			throw new IllegalArgumentException("Unknown Roman Numeral");
+		}
+		return mMap.get(inRNum);
+	}
+	public Integer evalToDecimal(ArrayList<String> inRNums) {
```

Comment:
Should this by List<Char> or String instead of ArrayList<String>?

Reply:
updated to list